assets/js/leaves.js

```diff
@@ -130,24 +171,17 @@ class Leaf {
     }
   }
 
-  static selectLeafByHash() {
-    if (location.hash.startsWith("#")) {
-      let leaf = document.querySelector(
-        `path[data-id=${location.hash.slice(1)}]`
-      );
-      // if hash id corresponds to a leaf, select it
-      if (leaf != undefined) {
-        // simulate click event, sending leaf as event target
-        Leaf.selectLeaf({ target: leaf });
-      }
-    }
-  }
-
   static closeLeafDetails() {
     const panel = document.querySelector("#leaf-details");
-    Leaf.deselectAll();
     panel.parentElement.classList.remove("show-details");
     panel.parentElement.classList.add("closed");
+    Leaf.setCurrentLeaf(undefined);
+    Leaf.closeTag();
+  }
+
+  static closeTag() {
+    // unset current tag and then call updateSelection
+    Leaf.setCurrentTag(undefined);
```

Comment:
Looks kind of weird to call these functions with `undefined`, and shouldn't be necessary. I think we get `undefined` if we don't pass a parameter at all - see [mdn on js default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters). Or we could set a default value if that looks cleaner / more readable to you (or if it's easier to test).

```suggestion
    Leaf.setCurrentLeaf();
    Leaf.closeTag();
  }

  static closeTag() {
    // unset current tag and then call updateSelection
    Leaf.setCurrentTag();
```